### PR TITLE
[#17]Configure devise for users

### LINKED ISSUE #17: Devise User Model

### PR DESCRIPTION
**Pull request by bransynluther**
# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- I have made corresponding changes to the documentation,
- I have added tests that prove my fix is effective or that my feature works,
- New and existing unit tests pass locally with my changes ("bundle exec rake"),
- Title include "WIP" if work is in progress.

-->

Resolves #17 

### Description
This PR Sets up devise for users. First, ran rails generate devise:install and then
added the mailcatcher gem. Added mailcatcher configurations to the
/config/environments/developmnent.rb to set up the mailer port and host.
The mailer port and host are assigned by using ENV files located in the
new .env file. An .env.example was also created so that anyone can
figure out what variables need to be set. Then this change used generated
controllers and views for devise user.
Next this creates the /users/views for edit and new. It also updates
the migrations for user devise model. Next Generate views for users/session,
and update routes to use /users/sessions. This is done because we are
overriding the default devise controllers.
The views for users session include sign in, gives the route to
sign out and delete the session, and also utilizes the navigation bar
to handle if the drop down says sign in or sign out based on the current
users state.
Guide questions:
  - What motivated this change (if not already described in an issue)?
  - What alternative solutions did you consider?
  - What are the tradeoffs for your solution?
   
List any dependencies that are required for this change. (gems, js libraries, etc.)
gems - mailcatcher
make copy of the .env.example as .env

Include anything else we should know about. -->

### Type of change

* New feature (non-breaking change which adds functionality)
* Breaking change (fix or feature that would cause existing functionality to not 
work as expected)

### How Has This Been Tested?

1. Added and removed .env filed to ensure that the .fetch defaults are working correctly.
2. Devise tests?

### Screenshots

image

### CODE REVIEW

**Review by gabezurita — COMMENTED** [review]
Great job!

File: app/views/devise/shared/_links.html.erb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+<%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
+  <%= link_to "Forgot your password?", new_password_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.confirmable? && controller_name != 'confirmations' %>
+  <%= link_to "Didn't receive confirmation instructions?", new_confirmation_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.lockable? && resource_class.unlock_strategy_enabled?(:email) && controller_name != 'unlocks' %>
```
Comment (maebeale) [inline app/views/devise/shared/_links.html.erb:13]: if we're not using lockable, we don't need these links. i personally like the lockable, but am open to not using it!

File: app/views/users/shared/_links.html.erb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+<%- if controller_name != 'sessions' %>
+  <%= link_to "Log in", new_session_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
+  <%= link_to "Forgot your password?", new_password_path(resource_name) %><br />
+<% end %>
+
+<%- if devise_mapping.confirmable? && controller_name != 'confirmations' %>
```
Comment (maebeale) [inline app/views/users/shared/_links.html.erb:9]: same here re confirmable and lockable

File: app/views/devise/shared/_links.html.erb (hunk 1 above)
Comment (bransynluther) [reply app/views/devise/shared/_links.html.erb:13]: My bad! I agree with you and think lockable should be included! I will create an issue and pr to address this.

File: app/views/users/shared/_links.html.erb (hunk 2 above)
Comment (bransynluther) [reply app/views/users/shared/_links.html.erb:9]: I will address this in the same issue and PR mentioned in the previous comment!

**Pull request merged**